elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryEngine.java

```diff
@@ -82,11 +80,24 @@ public SQLQueryEngine(MetaDataStore metaDataStore,
     public SQLQueryEngine(MetaDataStore metaDataStore,
                     com.yahoo.elide.contrib.dynamicconfighelpers.compile.ConnectionDetails defaultConnectionDetails,
                     Map<String, com.yahoo.elide.contrib.dynamicconfighelpers.compile.ConnectionDetails> detailsMap) {
-        this(metaDataStore, defaultConnectionDetails);
-        detailsMap.forEach((name, details) -> {
-            this.connectionDetailsMap.put(name, new ConnectionDetails(details.getDataSource(),
-                            SQLDialectFactory.getDialect(details.getDialect())));
-        });
+        if (defaultConnectionDetails == null) {
```

Comment:
I would just put google preconditions that defaultConn cannot be null nor can detailsMap.

Reply:
updated